dart/lib/src/sentry_isolate.dart

```diff
@@ -0,0 +1,95 @@
+import 'dart:async';
+import 'dart:isolate';
+
+import 'package:meta/meta.dart';
+
+import '../sentry.dart';
+
+/// Record isolate errors with the Sentry SDK.
+extension SentryIsolate on Isolate {
+
+  /// Calls [addErrorListener] with an error listener from the Sentry SDK. Store
+  /// the returned [RawReceivePort] if you want to remove the Sentry listener
+  /// again.
+  ///
+  /// Since isolates run concurrently, it's possible for it to exit before the
+  /// error listener is established. To avoid this, start the isolate paused,
+  /// add the listener and then resume the isolate.
+  RawReceivePort addSentryErrorListener() {
```

Comment:
To make it testable, we could take the `hub` as an optional parameter and use `HubAdapter` by default if none provided.